modules/augment/src/transform.cpp

```diff
@@ -0,0 +1,149 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+Transform::Transform() {}
+
+Transform::~Transform() {}
+
+void Transform::image(InputArray _src, OutputArray _dst)
+{
+    _src.copyTo(_dst);
+}
+
+Point2f Transform::point(const Point2f& src)
+{
+    return src;
+}
+
+Rect2f Transform::rectangle(const Rect2f& src)
+{
+    float x1 = src.x,
+        y1 = src.y,
+        x2 = src.x + src.width,
+        y2 = src.y + src.height;
+
+    Point2f tl(x1, y1);
+    Point2f bl(x1, y2);
+    Point2f tr(x2, y1);
+    Point2f br(x2, y2);
+
+    Point2f tl_transformed = this->point(tl);
+    Point2f bl_transformed = this->point(bl);
+    Point2f tr_transformed = this->point(tr);
+    Point2f br_transformed = this->point(br);
+
+    float x1_transformed = std::min({ tl_transformed.x, bl_transformed.x, tr_transformed.x, br_transformed.x });
+    float y1_transformed = std::min({ tl_transformed.y, bl_transformed.y, tr_transformed.y, br_transformed.y });
+    float x2_transformed = std::max({ tl_transformed.x, bl_transformed.x, tr_transformed.x, br_transformed.x });
+    float y2_transformed = std::max({ tl_transformed.y, bl_transformed.y, tr_transformed.y, br_transformed.y });
+
+    Rect2f output_box(Point2f(x1_transformed ,y1_transformed),  Point2f(x2_transformed ,y2_transformed));
+    return output_box;
+}
+
+Rect2f Transform::basicRectangle(const Rect2f& src)
```

Comment:
How this method differs from `Transform::rectangle`? When we need to use `basicRectangle` and when `rectangle`?

Reply:
rectangle() transforms the 4 corners and then calculates the new top left and bottom right even if the rectangle was deformed, that's why it's the general case, but basicRectangle is much faster because it only transforms the top left and bottom right and it depends on the fact that the rectangle shape will not change

Reply:
I think that we implemented specific rectangle() for each transformation so can I delete the basicRectangle() ?